src/main/python/server.py

```diff
@@ -420,342 +66,41 @@ def request_upload_file_index():
-        ),
-    error_y = dict(
-           type = 'data',
-           array = newError_rate,
-           visible = True
-        ),
-    error_z = dict(
-           type = 'data',
-           array = error,
-           visible = True
-        ),
-    marker=dict(
-        size=5,
-        color=newAmplitude,                # set color to an array/list of desired values
-        colorscale='Viridis',   # choose a colorscale
-        colorbar = dict(title = 'Amplitude<br>Map',thickness=20,len=1),
-        opacity=0.8
-        )
-    )
-
-    data3 = [trace3]
-
-    layout3 = go.Layout(
-        title= "Dymanic Spectrutm",
-    )
-
-
-    trace4 = dict(
-        type = 'scatter',
-        x = newTime,
-        y = newRate,
-        mode="markers",
-        marker=dict(
-        size=6,
-        color=newAmplitude,                # set color to an array/list of desired values
-        colorscale='Viridis',   # choose a colorscale
-        colorbar = dict(title = 'Amplitude<br>Map',thickness=20,len=1),
-        opacity=0.8
-        )
-    )
-
-    layout4 = dict(
-         title = '',
-         hovermode= 'closest',
-         xaxis = dict(
-             title = 'Time',
-             #range = [start_time, end_time],
-             titlefont = dict(
-                 family = 'Courier New, monospace',
-                 size = 18,
-                 color = '#7f7f7f'
-            )
-         ),
-         yaxis=dict(
-             title='Frequency',
-             #range=[start_count, end_count],
-             titlefont = dict(
-                 family = 'Courier New, monospace',
-                 size = 18,
-                 color = '#7f7f7f'
-             )
-         ),
-    )
-
-    start_time_int = min(Time)
-    end_time_int = max(Time)+1
-    start_count_int = min(Rate)
-    end_count_int = max(Rate)+1
-    start_color1_int = min(color1)
-    end_color1_int = max(color1)+1
-    start_color2_int = min(color2)
-    end_color2_int = max(color2)+1
-
-
-    fig=[]
-    fig1 = dict(data = [trace1], layout = layout1)
-    fig2 = dict(data = [trace2], layout = layout2)
-    fig3 = go.Figure(data=data3, layout=layout3)
-    fig4 = dict(data = [trace4], layout = layout4)
-
-    fig.append(fig1);
-    fig.append(fig2);
-    fig.append(start_time_int);
-    fig.append(end_time_int);
-    fig.append(start_count_int);
-    fig.append(end_count_int);
-    fig.append(start_color1_int);
-    fig.append(end_color1_int);
-    fig.append(start_color2_int);
-    fig.append(end_color2_int);
-    fig.append(fig3);
-    fig.append(fig4);
-
-    return fig
-
 @app.route('/')
-def my_form():
-    logging.debug("I am here")
-    return render_template("welcome.html")
+def root():
+    logging.debug("Root page requested!")
+    return render_template("master_page.html")
 
+#Setting error handler
+def http_error_handler(error):
+    return render_template("error.html", error=error ), error
 
+for error in (400, 401, 403, 404, 500): # or with other http code you consider as error
```

Comment:
According to this book on Flask that I have error 405 is "method not allowed" and is also not uncommon. But actually I would prefer not to register all possible error codes one by one, and instead just register one generic error handler that works for any response code that is an error (any code between 400 and 599). I'm not sure Flask support this though, the book doesn't mention it.